test/Conversion/tritongpu_to_llvm.mlir

```diff
@@ -1,4 +1,7 @@
-// RUN: triton-opt %s -split-input-file --allocate-shared-memory-nv --convert-triton-gpu-to-llvm 2>/dev/null | FileCheck %s --dump-input-context 20
+// RUN: triton-opt %s -split-input-file \
+// RUN:   --allocate-shared-memory-nv --convert-triton-gpu-to-llvm \
+// RUN:   --reconcile-unrealized-casts 2>/dev/null \
```

Comment:
we have quite a few tests that call `--convert-triton-gpu-to-llvm` why changing this is one in particular? 
We are not calling this pass in our python pass manager so either is is unnecessary or we may have some problems? Or is it just getting fixed by canonicalizer? 
I wonder if we should just make sure we reconcile unrealized casts at the end of our conversion passes instead

Reply:
Yes, probably they are being cleaned up by some canonicalization and/or DCE pass or similar. It could also be that some test allow for additional ops because they use CHECK and not CHECK-NEXT. Finally, downstream users and tests may not have updated to that version of LLVM yet.

In any case, yes, making the reconciliation part of the conversions may be a good way to ensure that that always happens. Maybe it's also something to ask or discuss upstream; sounds cumbersome if everybody needs to do this now...

Reply:
Yes, exactly. It breaks with the change linked in the PR description.